src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java

```diff
@@ -248,19 +249,40 @@ else if ( !ArtifactUtils.isSnapshot( project.getVersion() ) && altReleaseDeploym
         {
             getLog().info( "Using alternate deployment repository " + altDeploymentRepo );
 
-            Matcher matcher = ALT_REPO_SYNTAX_PATTERN.matcher( altDeploymentRepo );
+            Matcher matcher = ALT_LEGACY_REPO_SYNTAX_PATTERN.matcher( altDeploymentRepo );
 
-            if ( !matcher.matches() )
+            if ( matcher.matches() )
             {
-                throw new MojoFailureException( altDeploymentRepo, "Invalid syntax for repository.",
-                                                "Invalid syntax for alternative repository. Use \"id::url\"." );
+                getLog().warn( "Legacy form of alternate deployment parameter used, update parameter to match plugin version." );
```

Comment:
Idea: Maybe move this log below, and use `id` and `url` to show end user to what it should update. And not show this very same warn if legacy layout is NOT `"default"` but fail Mojo with exception...

Reply:
I agree, the message itself is partially not helpful. Indicate what is required.

Reply:
This message hides a lot of information that was gathered before. At this point, the plugin knows that `id::layout::url` was used (but expects `id::url`, of course). The message should reflect that, not just by saying `you used the old notation, use the new one. RTFM`, but more like `you used the 2.x notation id::layout::url, 3.x expects id::url`.
This would give not just a hint to whats wrong, but provide a solution... To me, this is the difference of a good plugin to a great one :sunglasses:
With this, it would be totally fine (and expected) to fail with the message, not continuing the build 'magically' if the layout is `default`.

Reply:
Will rework the PR then as follows:
* Mojo will first match LEGACY pattern, 
* if matched, build will fail, but the message will be saying what the problem is (legacy notation) and provide solution (modern notation), will not "magically" continue if layout is `default`
* continue with current pattern... 

This order (legacy then modern) is needed, as we saw, that modern pattern "matches" legacy notation, but messes up repositoryId (ends up w/ `serverId::default` and fails to find serverId in settings). This way, we exclude that possibility to have 2 (legacy), 3 (plain wrong) and more `::`s in string, as all those will be caught by legacy pattern (and in that case would provide bad "solution".

Reply:
Very nice! As I have said it is all about greedy regex....

Reply:
... now only to figure out on which computer I created this PR :laughing: